pkg/cosign/sign.go

```diff
@@ -75,3 +79,38 @@ type Identity struct {
 type Image struct {
 	DockerManifestDigest string `json:"Docker-manifest-digest"`
 }
+
+type Signer interface {
```

Comment:
Just so I understand - what other types of signer might there be? Is it per algorithm (ecdsa vs. rsa, etc.) or per "strategy" (key from disk, KMS, ephemeral, etc.)?

I mostly ask because I'm hoping we can avoid ever having to support multiple "algorithms" at the same time for signing, but could definitely see allowing for multiple types during verification.

Reply:
> Is it per algorithm (ecdsa vs. rsa, etc.) or per "strategy" (key from disk, KMS, ephemeral, etc.)?

I was thinking both, but I'm also happy to limit things to ECDSA+SHA256 for now. Until generics are a thing, and possibly even after then, it's hard to have a middle ground between "this thing" and "anything", which is presumably why the core crypto libs just gave up.

Reply:
I'm not sure the core crypto portion would be the right level to enforce opinionated-ness as to the forms of the signed payloads. I think we can just copy the web in this respect-- declare what crypto algos we accept but let the payload itself be a black box more-or-less

Reply:
We can consider fulcio and cosign somewhat independent, but we'll also want to nail down exactly what (if anything) fulcio should prescribe, as well, since that'll guide what we do for cosign

Reply:
Yup, agree everywhere!